paraatm/simulation_method/vcas.py

```diff
@@ -0,0 +1,337 @@
+            'data_file' : srt
+                directory to actual trajectory data as .csv file
+            'sim_time' : int/float
+                total simulation duration
+        """
+        self.fp_file = cfg['fp_file']
+        self.mfl_file = cfg['mfl_file']
+        self.cmd_file = cfg['cmd_file']
+        self.sim_time = cfg['sim_time']
+        self.track_file = cfg['data_file']
+        self.real = pd.read_csv(cfg['data_file'])
+
+        NatsEnvironment.start_jvm(nats_home=None)
+        self.NATS_SIMULATION_STATUS_PAUSE = NatsEnvironment.get_nats_constant('NATS_SIMULATION_STATUS_PAUSE')
+        self.NATS_SIMULATION_STATUS_ENDED = NatsEnvironment.get_nats_constant('NATS_SIMULATION_STATUS_ENDED')
+
+        natsStandalone = NatsEnvironment.get_nats_standalone()
+
+        self.simulationInterface = natsStandalone.getSimulationInterface()
+
+        self.entityInterface = natsStandalone.getEntityInterface()
+        self.controllerInterface = self.entityInterface.getControllerInterface()
+        self.pilotInterface = self.entityInterface.getPilotInterface()
+
+        self.environmentInterface = natsStandalone.getEnvironmentInterface()
+
+        self.equipmentInterface = natsStandalone.getEquipmentInterface()
+        self.aircraftInterface = self.equipmentInterface.getAircraftInterface()
+
+        if self.simulationInterface is None:
+            natsStandalone.stop()
+            raise RuntimeError("Can't get SimulationInterface")
+
+        self.simulationInterface.clear_trajectory()
+
+    def command_from_file(self):
+        """
+        extract commands related to altitude change
+
+        Returns
+        -------
+        Pandas DataFrame
+        """
+        text = pd.read_csv(self.cmd_file)
+        text = text.astype(object).replace(np.nan, 'None')
+        command_column = ['Timestamp', 'acID', 'action', 'altitude']
+        command_maintain = pd.DataFrame(columns=command_column)
+        for ind in range(len(text)):
+            line = text.loc[[ind]]
+            if 'maintain' in line['action1'].values[0] and line['new_alt'].values[0] != 'None':
+                timestamp = line['Timestamp'].values[0]
+                temp_c = [timestamp, line['acid'].values[0], line['action1'].values[0], line['new_alt'].values[0]]
+                command_maintain = command_maintain.append(pd.Series(temp_c, index=command_column), ignore_index=True)
+
+        cmd_maintain = command_maintain.sort_values(['Timestamp', 'acID'], ascending=True).reset_index(
+            drop=True)
+
+        return cmd_maintain
+
+    def get_start_time(self):
+        """
+        Extract simulation start time from input flight plan file into self variable
+        """
+        f = open(self.fp_file, 'r')
+        lines = f.readlines()
+        starttime = 999999999999
+        for x in lines:
+            if 'TRACK_TIME' in x:
+                ttemp = float(x[11:])
+                starttime = min(starttime, ttemp)
```

Comment:
This file is opened but never closed.  The best practice is to use a context manager (`with`), which guarantees that the file closes:
```python
        with open(self.fp_file, 'r') as f:
            lines = f.readlines()
            starttime = 999999999999
            for x in lines:
                if 'TRACK_TIME' in x:
                    ttemp = float(x[11:])
                    starttime = min(starttime, ttemp)
```